### PR TITLE
Sub-event splitting

### PR DESCRIPTION
**Pull request by nkx111**
This is related to issue #7. Uses relative timestamp for track grouping. Pipeline should be OK. Still needs verification with multiple simulaiton setup.

### CODE REVIEW

File: src/EventAction.cc [hunk 1]
```diff
@@ -284,35 +287,80 @@ void EventAction::ReOrderTrackIds(Int_t subId) {
...
     Int_t nTracks = restG4Event->GetNumberOfTracks();
+    Double_t timeDelay = restG4Metadata->GetSubEventTimeDelay();  // in unit us
 
-    Double_t timeDelay = restG4Metadata->GetSubEventTimeDelay() * REST_Units::s;
-
-    vector<Double_t> fTrackTimestampList;
-    fTrackTimestampList.clear();
-
+    // reorder tracks
+    std::map<int, TRestG4Track*> tracks;
```
Comment (jgalan) [inline src/EventAction.cc:295]: How it is this compiling? Should it be `TRestGeant4Track` ... there is a kind of dictionary?

**Comment by jgalan** [comment]
So, the validation pipeline for example 08 was not affected by the changes?

File: src/EventAction.cc (hunk 1 above)
Comment (nkx111) [reply src/EventAction.cc]: I forgot to rename it when copying the code from v2.2

**Comment by nkx111** [comment]
> So, the validation pipeline for example `07_FullChainDecay` was not affected by the changes?

Yes. However, the result entries seems to be different in `09_Pb210_Shielding`

**Comment by jgalan** [comment]
> > So, the validation pipeline for example `07_FullChainDecay` was not affected by the changes?
> 
> Indeed, `07_FullChainDecay` is OK. However, the result entries seems to be different in `09_Pb210_Shielding`

Yes, but I am surprised, at issue #7 the entries at absolute time splitting would be 13, but should not be 14 with relative splitting?

The problem with the last example is a different origin, probably connected with Geant4 version, we are looking at it with @lobis

File: src/EventAction.cc (hunk 1 above)
Comment (jgalan) [reply src/EventAction.cc]: Ok, but then I dont understand how commit e76e69f5 undergoes compilation and passes the pipeline at build stage.

https://gitlab.cern.ch/rest-for-physics/restg4/-/jobs/13684941

**Comment by nkx111** [comment]
It really strange. Maybe the previous build is not cleaned up?
I tested on my side and there is 14 kinds of isotopes being stored, and leading to a failed validation.

```
Daughter isotopes: 14
Bi210 Bi214 Pa234[73.920] Pb210 Pb214 Po210 Po214 Po218 Ra226 Rn222 Th230 Th234 U234 U238
Wrong number of isotopes found!
(int) 13
```

**Comment by jgalan** [comment]
Ok, the problem is that `master` branch was being pulled

**Pull request merged**